ext/gvl_tracing_native_extension/gvl_tracing.c

```diff
@@ -207,22 +281,28 @@ static void render_event(const char *event_name) {
   );
 }
 
-static void on_thread_event(rb_event_flag_t event_id, UNUSED_ARG const rb_internal_thread_event_data_t *_unused1, UNUSED_ARG void *_unused2) {
+static void on_thread_event(rb_event_flag_t event_id, const rb_internal_thread_event_data_t *event_data, UNUSED_ARG void *_unused2) {
+  thread_local_state *state = GT_EVENT_LOCAL_STATE(event_data, event_id & (
+        RUBY_INTERNAL_THREAD_EVENT_STARTED | RUBY_INTERNAL_THREAD_EVENT_RESUMED));
```

Comment:
Minor: May be worth documenting here that these two events are guaranteed to be called from the thread itself (and that's why we're filtering here to only those)

Reply:
Actually, these two are the one that are guaranteed to hold the GVL. That is why the are singled out here.